src/main/js/controls/Graph/helpers/helpers.js

```diff
@@ -419,6 +452,45 @@ const createVGridHandler = (gridSVG, axis, style, config) => {
         .classed(style, true)
         .call(translateVerticalGrid(axis, config));
 };
+/**
+ * calculates the character limit of label with respect to axis size
+ *
+ * @private
+ * @param {string} config - config object derived from input JSON
+ * @returns {object} character limit for both and y-axis
+ */
+const getAxesLabelCharacterLimits = (config) => ({
+    xAxisLimit: (3 * getXAxisWidth(config)) / 22,
+    yAndY2AxisLimit: config.height / 8
+});
+/**
+ * Adds onClick event for each tick.
+ * Criteria:
+ *  * Text needs to have a valid function for onClick
+ *
+ * @private
+ * @param {d3.selection} canvasSVG - d3 selection node of canvas svg
+ * @param {string} className - class name of axis
+ * @param {object} axisObj - Axis object provided by the consumer
+ * @param {number} charLimit - character limit of label in axis
+ * @returns {object} d3 svg path
+ */
+const addLabelEventHandler = (canvasSVG, className, axisObj, charLimit) =>
+    canvasSVG.selectAll(`.${className} text`).each(function (displayVal) {
+        if (
+            !canvasSVG.selectAll(`.${className} text`).text() ||
+            !utils.isFunction(axisObj.onClick)
+        ) {
+            return;
+        }
```

Comment:
here also can you add a comment?

Reply:
[fixed](https://github.com/cerner/carbon-graphs/pull/242/files#diff-608946a83d2c753fa2af8bbb05dc72d2R481)